src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -2155,6 +2156,55 @@ EnsurePartitionMetadataIsSane(char distributionMethod, int colocationId,
 						errmsg("Metadata syncing is only allowed for valid "
 							   "colocation id values.")));
 	}
+	else if (colocationId != INVALID_COLOCATION_ID &&
+			 distributionMethod == DISTRIBUTE_BY_HASH)
+	{
+		int count = 1;
+		List *targetColocatedTableList =
+			ColocationGroupTableList(colocationId, count);
+
+		/*
+		 * If we have any colocated hash tables, ensure if they share the
+		 * same distribution key properties.
+		 */
+		if (list_length(targetColocatedTableList) == 1)
```

Comment:
```suggestion
		if (list_length(targetColocatedTableList) >= 1)
```

Reply:
We pass `count = 1` to  `ColocationGroupTableList`, that's why I used `== 1`. But, yes, lets not have such a dependency. I'll update.

(ps: I don't like committing the suggestion to remote branches as it complicates my git workflow. Instead, I manually locally)

Reply:
> (ps: I don't like committing the suggestion to remote branches as it complicates my git workflow. Instead, I manually locally)

Typing `git pull` is an issue?

Reply:
Yes, I'm not comfortable with `git pull` for my own branches :)